src/library_syscall.js

```diff
@@ -114,10 +114,12 @@ var SyscallsLibrary = {
       }
       return 0;
     },
-    doDup: function(path, flags, suggestFD) {
+    doDup: function(stream, suggestFD) {
       var suggest = FS.getStream(suggestFD);
       if (suggest) FS.close(suggest);
-      return FS.open(path, flags, 0, suggestFD, suggestFD).fd;
+
+      FS.streams[suggestFD] = stream;
```

Comment:
Yes, new implementation is EXACTLY what is required. duplicated file descriptor SHOULD refer to same opened files/sockets/terminals. Everything should be shared except O_CLOXEC (since O_CLOEXEC is file descriptor property, it opposed to "opened file" properties).